### PR TITLE
Add multiple instance sample

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Adds a fake vendor library creating its own named Datadog instance.

### Motivation

The goal is to verify and showcase how library maintainers can use named Datadog instance.

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: sample/kotlin/src/main/kotlin/com/datadog/android/sample/webview/WebViewModel.kt [hunk 1]
```diff
@@ -7,11 +7,11 @@ package com.datadog.android.sample.webview
 
 import androidx.lifecycle.ViewModel
 import com.datadog.android.sample.BuildConfig
-import com.datadog.android.sample.server.LocalServer
+import com.datadog.android.vendor.sample.LocalServer
 
 internal class WebViewModel : ViewModel() {
 
-    val localServer = LocalServer()
+    val localServer = com.datadog.android.vendor.sample.LocalServer()
```
Comment (0xnm) [inline sample/kotlin/src/main/kotlin/com/datadog/android/sample/webview/WebViewModel.kt:14]:
```suggestion
    val localServer = LocalServer()
```

**Review by 0xnm — APPROVED** [review]

**Review by 0xnm — DISMISSED** [review]

File: sample/vendor-lib/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    implementation(libs.androidXMultidex)
+
+    // Ktor (local web server)
+    implementation(libs.bundles.ktor)
+
+    // Multithreading
+    implementation(libs.rxJava3)
+    implementation("com.squareup.retrofit2:adapter-rxjava3:2.9.0")
+    implementation(libs.rxJava3Android)
+    implementation(libs.coroutinesCore)
```
Comment (0xnm) [inline sample/vendor-lib/build.gradle.kts:92]: are they needed here? seems like there is no use of RxJava or Coroutines in this module. Maybe OkHttp below is not needed as well.

File: sample/vendor-lib/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+            excludes += "META-INF/*"
+        }
+    }
+}
+
+dependencies {
+
+    implementation(project(":features:dd-sdk-android-logs"))
+    implementation(project(":features:dd-sdk-android-trace"))
+    implementation(project(":integrations:dd-sdk-android-okhttp"))
```
Comment (0xnm) [inline sample/vendor-lib/build.gradle.kts:78]: these 2 seems not to be used as well.

**Review by 0xnm — APPROVED** [review]
lgtm! just one comment about dependencies.

**Review by mariusc83 — APPROVED** [review]

**Pull request merged**